simapp/app.go

```diff
@@ -189,7 +192,7 @@ func NewSimApp(
 	app.CrisisKeeper = crisis.NewKeeper(
 		app.subspaces[crisis.ModuleName], invCheckPeriod, app.SupplyKeeper, auth.FeeCollectorName,
 	)
-	app.UpgradeKeeper = upgrade.NewKeeper(skipUpgradeHeights, keys[upgrade.StoreKey], app.cdc)
+	app.UpgradeKeeper = upgrade.NewKeeper(skipUpgradeHeights, keys[upgrade.StoreKey], app.cdc, viper.GetString(flags.FlagHome))
```

Comment:
The viper dep should not be used here. We need to figure out a way to provide the home value app constructor instead.

Something we need to address in another PR is really cleaning up the app constructor. We should use variadic option functions maybe.

Reply:
This might help: https://github.com/cosmos/cosmos-sdk/issues/5548

Reply:
Yes, we should address that issue. But I would suggest doing it separately to keep this PR as lean as possible. For the time being, simply have the SimApp constructor take a new parameter or even introduce variadic options.